### ISSUE TITLE
New AWS YML configuration leads to rails & mongo throttling issues

### ISSUE THREAD

**Issue by rHorsey** [NREL/OpenStudio-server#448]
### General Issue / Feature Information

**Description**

In modest-scale analyses on AWS (~150 cores in the cluster) the web gui becomes unavailable, and mongo requests time out frequently after ~30 minutes. 

**Reproduction steps:**
1. Run a project that will keep several worker nodes busy for more than an hour
2. Wait 20 minutes
3. Refresh the web gui

**Actual outcome:**
Timeout error

**Expected outcome:**
Webpage loads

### Error on Remote Server Details

**AMI version being used**

2.7.0

**Server management log file**

Not useful in this case - instead refer to this [commit](https://github.com/NREL/OpenStudio-server/commit/e580bd413da02e3c5d8ee1f91383bdd0a9afcd26#diff-faf230be006717d296d00f2c4b3fce43) and the screenshot from @ejhw that will be attached momentarily from the server node. You will see more than 4 oscli / E+ threads, even though 12 cores should be reserved for use by the non-worker containers.

**Comment by ejhw** [comment]
![image](https://user-images.githubusercontent.com/1276021/49837161-ae700780-fd62-11e8-944e-8458cd15c8e0.png)

**Comment by ejhw** [comment]
```
version: '3'
services:
  db:
    image: localhost:5000/mongo
    ports:
      - "27017:27017"
    volumes:
      - dbdata:/data/db
    deploy:
      placement:
        constraints:
          - node.role == manager
      resources:
        reservations:
          cpus: "3"
  queue:
    image: redis
    ports:
      - "6379:6379"
    deploy:
      placement:
        constraints:
          - node.role == manager
      resources:
        limits:
          cpus: "1"
  web:
    image: localhost:5000/openstudio-server
    ports:
      - "8080:80"
      - "80:80"
      - "443:443"
    environment:
      - OS_SERVER_NUMBER_OF_WORKERS=134
      - MAX_REQUESTS=180
      - MAX_POOL=80
      - QUEUES=analysis_wrappers
    volumes:
      - osdata:/mnt/openstudio
    depends_on:
      - db
      - queue
    deploy:
      placement:
        constraints:
          - node.role == manager
      resources:
        reservations:
          cpus: "5"
    command: /usr/local/bin/start-server
  web-background:
    image: localhost:5000/openstudio-server
    environment:
      - OS_SERVER_NUMBER_OF_WORKERS=134
      - QUEUES=background,analyses
    volumes:
      - osdata:/mnt/openstudio
    depends_on:
      - db
      - web
      - queue
    deploy:
      placement:
        constraints:
          - node.role == manager
      resources:
        limits:
          cpus: "1"
    command: /usr/local/bin/start-web-background
  worker:
    image: localhost:5000/openstudio-server
    environment:
      - QUEUES=simulations
      - COUNT=1
    volumes:
      - /mnt/openstudio
    depends_on:
      - web
      - web-background
      - db
      - queue
      - rserve
    deploy:
      resources:
        limits:
          cpus: "1"
    command: /usr/local/bin/start-workers
  rserve:
    image: localhost:5000/openstudio-rserve
    volumes:
      - osdata:/mnt/openstudio
    depends_on:
      - web
      - web-background
      - db
    deploy:
      placement:
        constraints:
          - node.role == manager
      resources:
        limits:
          cpus: "1"
volumes:
  osdata:
    external: true
  dbdata:
    external: true
networks:
  default:
    driver: overlay
    ipam:
      driver: default
      config:
        - subnet: 172.28.0.0/16
```

**Comment by rHorsey** [comment]
Note that this makes multi-node cluster unstable, negating the point of docker swarm. A reasonable place to catch the error this is attempting to address (the swarmkit scheduler is unable to satisfy resource requirements) is in the PAT gui settings for creating clusters.

**Comment by ejhw** [comment]
This is holding me up on a series of runs I need to do for a university partner, so if someone could make a custom AMI for me that would be much appreciated.

**Comment by brianlball** [comment]
There is only 8 reserved cores in that yml file, 3 for db and 5 for web, not 12.

**Comment by brianlball** [comment]
@rHorsey @ejhw We need more information and what solution are you proposing?  Its not clear that the referenced commit is a problem since that has been working since July (fixed users not being able to start servers with smaller resources) and we have other anecdotal evidence of other users not having a problem with larger clusters.

If you want to propose changes to the docker-compose.yml file, I suggest you start your existing cluster, ssh into the server and edit that file to your changes and then restart the swarm.  I tested that this will work this morning on AWS and you dont need new AMIs for that.

I'm sure aware but the commands to re-init the swarm after you edit the yml file are
>docker stack rm osserver
>docker stack deploy osserver –compose-file=$HOME/docker-compose.yml
>docker service scale osserver_worker=xx
 where xx is the number of workers you have.

**Comment by ejhw** [comment]
I understand that this commit fixes a problem for some users, but it causes a problem for other users. It makes it impossible to use multi-node clusters for the ResStock workflow that we support for multiple private companies and academic partners. 

We plan to move away from using PAT in February, but in the meantime, we need to support the ResStock workflow on PAT. Since we don't need PAT long term, my proposed solution is to make a custom _2.7.0-resstock_ AMI (with the referenced commit reverted) as a bandaid fix.

It isn't really acceptable to expect our users to ssh into servers to fix this issue.

**Comment by brianlball** [comment]
@ejhw I've sent you a PM on slack to discuss but nobody is suggesting your users use the ssh solution.  That is to triage the problem and determine what the fix should be.

**Comment by brianlball** [comment]
the 'limits' in the yml file were indeed causing lots of workers to be assigned to the server node.  @anyaelena created an AMI '2.7.1-largescale0' which has that change reverted.  

An issue was also created in PAT for addressing the related resource startup error for smaller servers by removing the 4cpu servers from possible selection here https://github.com/NREL/OpenStudio-PAT/issues/101.  That in combo with a revert on the docker swam config file should solve both issues.

**Comment by anyaelena** [comment]
This revert was merged and will be included in 2.7.2. @ejhw i heard you may still be seeing a similar issue.  If so, I'll collect details from you and open a new issue.
